subprojects/antlr/src/main/java/org/gradle/api/plugins/antlr/AntlrPlugin.java

```diff
@@ -40,10 +41,12 @@
 public abstract class AntlrPlugin implements Plugin<Project> {
     public static final String ANTLR_CONFIGURATION_NAME = "antlr";
     private final ObjectFactory objectFactory;
+    private final TaskDependencyFactory taskDependencyFactory;
 
     @Inject
-    public AntlrPlugin(ObjectFactory objectFactory) {
+    public AntlrPlugin(ObjectFactory objectFactory, TaskDependencyFactory taskDependencyFactory) {
```

Comment:
I think you can get rid of several of these changes that are just snaking this dependency through classes.

Because we're creating `DefaultAntlrSourceDirectorySet` via `ObjectFactory`, it will fill in anything that's an available service automatically. 

In places where we are not instantiating via ObjectFactory or Instantiator, you'll have to manually wire things.  

Here's an example: https://github.com/gradle/gradle/commit/2fb3c5bdfd8fe5681cde9666beb1be6e89403ede

Reply:
I believe some constructors have become `@Inject` just recently. I'll check those again. Thanks!

Reply:
✅ https://github.com/gradle/gradle/pull/22433/commits/b581cea8fbfd9dbd808369090d67dbb0b6fd132b

Reply:
Thanks @h0tk3y